### PR TITLE
ANN: improve annotator highlighting

### PR DESCRIPTION
**Pull request by Undin**
* correctly highlight module declarations, module and crate references
* provide API to register custom severity to simplify highlighting tests
* provide API to test only particular annotator
* use `info` severity to highlight using of unsafe code by unsafe annotator
* reorder annotator declarations in `plugin.xml` to launch faster annotators first

### CODE REVIEW

**Comment by Undin** [comment]
@garyttierney about annotator disabling
Since this PR will have been merged, it will be possible enable only particular annotator in tests

File: src/main/kotlin/org/rust/ide/annotator/RsHighlightingMutableAnnotator.kt [hunk 1]
```diff
@@ -66,16 +59,16 @@ class RsHighlightingMutableAnnotator : Annotator {
...
-            annotationText
-        ).textAttributes = key.textAttributesKey
+        val annotationSeverity = if (isUnitTestMode) key.testSeverity else MUTABLE_HIGHLIGHTING
+        holder.createAnnotation(annotationSeverity, target.textRange, null).textAttributes = key.textAttributesKey
     }
 
+    companion object {
+        private val MUTABLE_HIGHLIGHTING = HighlightSeverity(
+            "MUTABLE_HIGHLIGHTING",
+            HighlightSeverity.INFORMATION.myVal + 3
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsHighlightingMutableAnnotator.kt]: Why 3?

**Comment by vlad20012** [comment]
:+1: :+1: :+1:
bors r+

**Comment by Undin** [comment]
bors r=vlad20012

File: src/main/kotlin/org/rust/ide/annotator/RsHighlightingMutableAnnotator.kt (hunk 1 above)
Comment (Undin) [reply src/main/kotlin/org/rust/ide/annotator/RsHighlightingMutableAnnotator.kt]: Just some positive value to be greater than `HighlightSeverity.INFORMATION`
If I remember correctly it was inspired by [LOW_PRIORITY_HIGHLIGHTING](https://github.com/JetBrains/intellij-community/blob/master/python/src/com/jetbrains/python/validation/HighlightingAnnotator.java#L37) highlight severity in python plugin

**Comment by Undin** [comment]
bors r=vlad20012

**Pull request merged**